### PR TITLE
Tmedia 722 links bar block refactor

### PR DESCRIPTION
**Pull request by malavikakoppula**
Update links-bar block to use v2 architecture

- [TMEDIA-722](https://arcpublishing.atlassian.net/browse/TMEDIA-722)

## Acceptance Criteria

_copy from ticket_

Follow checklist https://arcpublishing.atlassian.net/wiki/spaces/TI/pages/3106243821/Converting+a+v1+block+to+v2+architecture 

match the live styles on the page in storybook in the news theme. 

This should include desktop and mobile styles

When you push to a wip in progress branch, you can look at the chromatic diff that we saw in the pr walkthroughs 

Use the v2 themes components. 

Should import the Link component in themes components 

## Test Steps

- Add test steps a reviewer must complete to test this PR

1. Checkout this branch `git checkout {branch name}`
2. Run fusion repo with linked blocks `npx fusion start -f -l {blocks to link}`
3. {Next ....}

### Before

_Example: When I clicked the search button, the button turned red._

[include screenshot or gif or link to video, storybook would be awesome]

### After

_Example: When I clicked the search button, the button turned green._

[include screenshot or gif or link to video, storybook would be awesome]

## Dependencies or Side Effects

_Examples of dependencies or side effects are:_

- Additional settings that will be required in the blocks.json
- Changes to the custom fields which will require users to reconfigure features
- Update to css framework or SDK
- Dependency on another PR that needs to be merged first

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

- [ ] Confirmed all the test steps a reviewer will follow above are working.
- [ ] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [ ] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [ ] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
        please explain why (so that we can fix it whenever it gets refactored).
- [ ] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

File: .storybook/mock-content/linksBar.js [hunk 1]
```diff
@@ -6,37 +6,37 @@ export const linksBarMock = (query) => {
 			children: [
 				{
 					_id: "id_1",
-					name: "test link 1",
```
Comment (JackHowa) [inline .storybook/mock-content/linksBar.js]: not sure I totally follow why this mock content was changed. Is something wrong with the existing text? I usually keep what's there so that it's easier to tell what's changed

File: blocks/footer-block/features/footer/default.jsx [hunk 2]
```diff
@@ -17,6 +17,9 @@ import {
 import { PrimaryFont } from "@wpmedia/shared-styles";
 import "./footer.scss";
 
+// switch to component link when footer converting to V2
+import Link from "./_children/link";
```
Comment (JackHowa) [inline blocks/footer-block/features/footer/default.jsx:20]: Looks like there were changes to the footer block? was this on purpose? We only want changes to the links bar block. can you revert these changes

File: blocks/links-bar-block/_index.scss [hunk 3]
```diff
@@ -0,0 +1,14 @@
+@use "@wpmedia/arc-themes-components/scss";
+
+.b-links-bar {
+	@include scss.block-components("links-bar");
+	@include scss.block-properties("links-bar");
+	&-menu {
```
[the remark this answers is not in the document]
Comment (JackHowa) [reply blocks/links-bar-block/_index.scss]: @malavikakoppula it looks like we need to run lint styles again

File: blocks/links-bar-block/features/links-bar/default.jsx [hunk 4]
```diff
@@ -35,19 +34,23 @@ const LinksBar = ({ customFields: { navigationConfig = {}, ariaLabel } }) => {
...
 			<nav
 				key={id}
-				className="links-bar"
+				className={BLOCK_CLASS_NAME}
 				aria-label={ariaLabel || phrases.t("links-bar-block.element-aria-label")}
 			>
 				{menuItems &&
 					menuItems.map((item, index) => (
-						<PrimaryFont as="span" className="links-menu" key={item._id}>
+						<span className={`${BLOCK_CLASS_NAME}-menu`} key={item.display_name}>
```
Comment (JackHowa) [inline blocks/links-bar-block/features/links-bar/default.jsx]: can you use the bem naming standard of using "__" rather than "-" for an element of a block http://getbem.com/naming/
```suggestion
						<span className={`${BLOCK_CLASS_NAME}__menu`} key={item.display_name}>
```

File: blocks/links-bar-block/features/links-bar/default.jsx [hunk 5]
```diff
@@ -35,19 +34,23 @@ const LinksBar = ({ customFields: { navigationConfig = {}, ariaLabel } }) => {
...
 							{item.node_type === "link" ? (
-								<Link href={item.url} name={item.display_name} />
+								<Link href={item.url} className={`${BLOCK_CLASS_NAME}-menu-link`} openInNewTab>
```
Comment (JackHowa) [inline blocks/links-bar-block/features/links-bar/default.jsx]: Same here. we want to use the element naming

Comment (JackHowa) [inline blocks/links-bar-block/features/links-bar/default.jsx]: is there a reason why some of these links are using "openInNewTab"?

File: blocks/links-bar-block/package.json [hunk 6]
```diff
@@ -8,10 +8,11 @@
   ],
```
Comment (JackHowa) [inline blocks/links-bar-block/package.json]: can you also update the author to be based on the checklist. and remove the contributors. Let's keep on using the checklist for updating the blocks https://arcpublishing.atlassian.net/wiki/spaces/TI/pages/3106243821/Converting+a+v1+block+to+v2+architecture

File: blocks/links-bar-block/package.json [hunk 7]
```diff
@@ -30,7 +31,8 @@
     "@arc-fusion/prop-types": "^0.1.5",
     "@wpmedia/engine-theme-sdk": "*",
     "@wpmedia/news-theme-css": "*",
```
Comment (JackHowa) [inline blocks/links-bar-block/package.json]: we want to remove the packages like news-theme-css that are not being used per the checklist. Shared styles should also be removed
```suggestion
```

File: .storybook/themes/news.scss [hunk 8]
```diff
@@ -912,3 +943,4 @@
 @use "../../blocks/story-carousel-block";
 @use "../../blocks/triple-chain-block";
 @use "../../blocks/video-player-block";
+@use "../../blocks/links-bar-block";
```
Comment (JackHowa) [inline .storybook/themes/news.scss]: can you add the links bar to the list based on alphabetical order. it's not said in the docs but I use the sort ascending tool on vs code

**Review by JackHowa — CHANGES_REQUESTED** [review]
Let's make sure we're following the checklist for standards. It's sometimes easy to looking at other code and accidentally making changes https://arcpublishing.atlassian.net/wiki/spaces/TI/pages/3106243821/Converting+a+v1+block+to+v2+architecture

Also, let's focus on using bem naming like we talked about. http://getbem.com/naming/

File: blocks/links-bar-block/_index.scss [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+		@include scss.block-components("links-bar-menu");
+		@include scss.block-properties("links-bar-menu");
+		a {
```
[the remark this answers is not in the document]
Comment (JackHowa) [reply blocks/links-bar-block/_index.scss:5]: fixed

File: blocks/links-bar-block/_index.scss (hunk 3 above)
[the remark this answers is not in the document]
Comment (JackHowa) [reply blocks/links-bar-block/_index.scss]: this should be fixed

File: blocks/share-bar-block/README.md [hunk 10]
```diff
@@ -1,10 +1,19 @@
 # `@wpmedia/share-bar-block`
 
-_This is the share bar block for the news theme, written as a functional component. Users can choose out of five options (email, facebook, pinterest, linkedin, twitter) which selection of share buttons will be displayed on articles. Clicking on each button will open up a new window._
+This is the share bar block for the news theme, written as a functional component. Users can choose from five options to display on the share bar:
```
Comment (JackHowa) [inline blocks/share-bar-block/README.md]: this should have been removed. it must have been re-added

File: .storybook/mock-content/linksBar.js [hunk 11]
```diff
@@ -36,7 +36,7 @@ export const linksBarMock = (query) => {
 					_id: "id_3",
 					node_type: "link",
 					url: "/",
-					display_name: "Link Text",
+					display_name: "Link Text 3",
```
Comment (JackHowa) [inline .storybook/mock-content/linksBar.js]: looks like there's still some changes to the display text here based on previous change 
```suggestion
					display_name: "Link Text",
```

File: blocks/footer-block/features/footer/default.jsx (hunk 2 above)
Comment (malavikakoppula) [reply blocks/footer-block/features/footer/default.jsx:20]: yes, this is on purpose to avoid the dependency on link-bar block

Comment (JackHowa) [reply blocks/footer-block/features/footer/default.jsx:20]: I'm working on the footer block and will fix this. it's ok for you to revert this code.

File: blocks/links-bar-block/features/links-bar/default.jsx [hunk 12]
```diff
@@ -35,19 +34,23 @@ const LinksBar = ({ customFields: { navigationConfig = {}, ariaLabel } }) => {
...
+								<Link href={item.url} className={`${BLOCK_CLASS_NAME}__menu-link`} openInNewTab>
+									{item.display_name}
+								</Link>
 							) : (
-								<Link href={item._id} name={item.name} />
+								<Link href={item._id} className={`${BLOCK_CLASS_NAME}__menu-link`}>
+									{item.name}
+								</Link>
 							)}
 							{content.children.length !== index + 1 && showSeparator ? "\u00a0 • \u00a0" : ""}
```
Comment (JackHowa) [inline blocks/links-bar-block/features/links-bar/default.jsx]: this can use the separator component in the arc themes components. this info is avilable in the figma https://www.figma.com/file/vWVGgZfZVBdSX7ACo3kmjr/Enhanced-Styling-Design-System?node-id=3507%3A34664 

https://github.com/WPMedia/arc-themes-components/tree/arc-themes-release-version-2.0.1/src/components/separator

Screen Shot 2022-06-03 at 10 21 31

Comment (malavikakoppula) [reply blocks/links-bar-block/features/links-bar/default.jsx]: test

Comment (ecarlisle) [reply blocks/links-bar-block/features/links-bar/default.jsx]: @malavikakoppula , @JackHowa : FYI, last week I noticed there were some inconsistent styling in Figma. I asked Kelly about this and she fixed it. We just need to make sure that we're updated with any changes. For example, the Figma design shows each link being underlined now.

File: blocks/footer-block/features/footer/default.jsx (hunk 2 above)
Comment (ecarlisle) [reply blocks/footer-block/features/footer/default.jsx:20]: @JackHowa : Will this affect the links appearing with `font-family: Abel`?
In Story book I'm seeing the link in a Serif font, not `$font-family-primary` (Inter) like the design in Figma.
Thanks!

File: .storybook/themes/news.scss [hunk 13]
```diff
@@ -939,6 +955,21 @@
...
+					text-align: center,
+					width: 100%,
+				),
+				links-bar-menu: (
+					padding: var(--global-spacing-2) 0,
+				),
+				links-bar-menu-link: (
+					font-size: var(--global-font-size-3),
+					color: var(--global-neutral-8),
+					line-height: var(--global-font-size-14),
```
Comment (ecarlisle) [inline .storybook/themes/news.scss]: Hi @malavikakoppula !
We should use the `var(--global-line-height)` tokens for specifying line-height values.

File: .storybook/themes/news.scss [hunk 14]
```diff
@@ -788,6 +788,22 @@
...
+					align-items: flex-start,
+					text-align: left,
+					width: 100%,
+				),
+				links-bar-menu: (
+					padding: var(--global-spacing-8) 0,
+				),
+				links-bar-menu-link: (
+					font-size: var(--global-font-size-3),
+					color: var(--global-neutral-8),
```
Comment (ecarlisle) [inline .storybook/themes/news.scss]: It looks like in Figma that the link color is `#000000`.
We can use the `var(--color-primary)` alias for this.
image

File: .storybook/themes/news.scss [hunk 15]
```diff
@@ -788,6 +788,22 @@
...
+					line-height: var(--global-line-height-3),
+					margin: var(--global-spacing-8) 0,
+					text-align: center,
+					text-decoration: none,
```
Comment (ecarlisle) [inline .storybook/themes/news.scss]: I'm asking about an inconsistency for this in Figma. The links appear to be underlined, but the typography properties say that text-decoration is none. 🤔 
I'm asking about that here:
https://www.figma.com/file/vWVGgZfZVBdSX7ACo3kmjr?node-id=3507:34664#206727951

File: .storybook/themes/news.scss [hunk 16]
```diff
@@ -939,6 +955,21 @@
...
+					align-items: flex-start,
+					text-align: center,
+					width: 100%,
+				),
+				links-bar-menu: (
+					padding: var(--global-spacing-2) 0,
+				),
+				links-bar-menu-link: (
+					font-size: var(--global-font-size-3),
+					color: var(--global-neutral-8),
```
Comment (ecarlisle) [inline .storybook/themes/news.scss]: From Figma it looks like the color is `#000000`, which is the `var(--color-primary)` alias.
That's the default color for a `<Link/>` component, so you probably don't have to specify a color.

https://github.com/WPMedia/arc-themes-feature-pack/blob/news-2.00/site-styles/daily-intelligencer/_index.scss#L358

File: .storybook/themes/news.scss [hunk 17]
```diff
@@ -939,6 +955,21 @@
...
+				links-bar: (
+					align-items: flex-start,
+					text-align: center,
+					width: 100%,
+				),
+				links-bar-menu: (
+					padding: var(--global-spacing-2) 0,
+				),
+				links-bar-menu-link: (
+					font-size: var(--global-font-size-3),
```
Comment (ecarlisle) [inline .storybook/themes/news.scss]: Looks like in Figma that the font size is of a link is `var(--global-font-size-4)`, which is the same as the `var(--body-font-size)` alias. You probably don't need to specify a font size here.

File: .storybook/themes/news.scss [hunk 18]
```diff
@@ -788,6 +788,22 @@
 				video-player-inline-video: (
 					gap: 0,
 				),
+				links-bar: (
+					align-items: flex-start,
+					text-align: left,
+					width: 100%,
+				),
+				links-bar-menu: (
+					padding: var(--global-spacing-8) 0,
```
Comment (ecarlisle) [inline .storybook/themes/news.scss]: I don't think the top or bottom padding is specified in the Figma.
I'm asking about it here as well.
https://www.figma.com/file/vWVGgZfZVBdSX7ACo3kmjr?node-id=3507:34664#206704565

File: .storybook/themes/news.scss [hunk 19]
```diff
@@ -788,6 +788,22 @@
...
+				),
+				links-bar-menu-link: (
+					font-size: var(--global-font-size-3),
+					color: var(--global-neutral-8),
+					line-height: var(--global-line-height-3),
```
Comment (ecarlisle) [inline .storybook/themes/news.scss]: In the Figma it looks like we're using var(--global-line-height-4) here:
image

**Review by ecarlisle — CHANGES_REQUESTED** [review]

File: .storybook/themes/news.scss (hunk 15 above)
Comment (malavikakoppula) [reply .storybook/themes/news.scss]: Brandon confirmed that it should be text-decoration: none

File: blocks/links-bar-block/features/links-bar/default.jsx [hunk 20]
```diff
@@ -35,19 +34,30 @@ const LinksBar = ({ customFields: { navigationConfig = {}, ariaLabel } }) => {
...
 							)}
-							{content.children.length !== index + 1 && showSeparator ? "\u00a0 • \u00a0" : ""}
-						</PrimaryFont>
+							{/* {content.children.length !== index + 1 && showSeparator ? "\u00a0 • \u00a0" : ""} */}
+							{content.children.length !== index + 1 && showSeparator ? (
+								<span className={`${BLOCK_CLASS_NAME}__menu-link-seperator`}>
+									<Separator />
+								</span>
+							) : (
+								""
```
Comment (JackHowa) [inline blocks/links-bar-block/features/links-bar/default.jsx:58]:
```suggestion
								null
```

File: blocks/links-bar-block/features/links-bar/default.jsx [hunk 21]
```diff
@@ -35,19 +34,30 @@ const LinksBar = ({ customFields: { navigationConfig = {}, ariaLabel } }) => {
...
+								</Link>
 							) : (
-								<Link href={item._id} name={item.name} />
+								<Link href={item._id} className={`${BLOCK_CLASS_NAME}__menu-link`}>
+									{item.name}
+								</Link>
 							)}
-							{content.children.length !== index + 1 && showSeparator ? "\u00a0 • \u00a0" : ""}
-						</PrimaryFont>
+							{/* {content.children.length !== index + 1 && showSeparator ? "\u00a0 • \u00a0" : ""} */}
```
Comment (JackHowa) [inline blocks/links-bar-block/features/links-bar/default.jsx]:
```suggestion
```

can you remove the commented out code

File: .storybook/themes/news.scss (hunk 15 above)
Comment (ecarlisle) [reply .storybook/themes/news.scss]: Fantastic! Thanks!

Comment (ecarlisle) [reply .storybook/themes/news.scss]: Fantastic! Thanks!

File: .storybook/themes/news.scss (hunk 18 above)
Comment (JackHowa) [reply .storybook/themes/news.scss]: Like we talked about these token names do not need the bem naming -- only the scss class names. You know what I'm talking about right with the difference between tokens and classnames? this is a naming standard that you'll see elsewhere in the code as well

Comment (JackHowa) [reply .storybook/themes/news.scss]: should be `links-bar-menu:`

File: .storybook/themes/news.scss [hunk 22]
```diff
@@ -822,6 +822,15 @@
 				video-player-inline-video: (
 					gap: 0,
 				),
+				links-bar: (
+					display: flex,
+					justify-content: center,
+					align-items: center,
+					gap: var(--global-spacing-2),
+				),
+				links-bar__menu-link: (
```
Comment (JackHowa) [inline .storybook/themes/news.scss]:
```suggestion
				links-bar-menu-link: (
```

For the token names, we've been using dashes to separate the names. I'm sorry if this isn't clear. Do you remember talking about the difference between tokens and scss classes? They don't have to be the same. The token names, by our convention, are separated by dashes and the scss classes use bem naming

File: .storybook/themes/news.scss [hunk 23]
```diff
@@ -976,6 +985,15 @@
 						),
 					),
 				),
+				links-bar: (
+					display: flex,
+					justify-content: center,
+					align-items: center,
+					gap: var(--global-spacing-2),
+				),
+				links-bar__menu-link: (
```
Comment (JackHowa) [inline .storybook/themes/news.scss]:
```suggestion
				links-bar-menu-link: (
```

File: blocks/links-bar-block/_index.scss [hunk 24]
```diff
@@ -0,0 +1,21 @@
+@use "@wpmedia/arc-themes-components/scss";
+
+.b-links-bar {
+	&__menu {
+		a {
+			@include scss.block-components("links-bar__menu-link");
+			@include scss.block-properties("links-bar__menu-link");
```
Comment (JackHowa) [inline blocks/links-bar-block/_index.scss]:
```suggestion
			@include scss.block-components("links-bar-menu-link");
			@include scss.block-properties("links-bar-menu-link");
```

Like I said in the previous comment, we want the scss classes to be bem naming. And we want the tokens to be separated by dashes, like other tokens. Can you apply this change to other tokens only

File: blocks/links-bar-block/features/links-bar/default.jsx [hunk 25]
```diff
@@ -35,19 +34,29 @@ const LinksBar = ({ customFields: { navigationConfig = {}, ariaLabel } }) => {
...
+							</span>
+							{content.children.length !== index + 1 && showSeparator ? (
+								<span className={`${BLOCK_CLASS_NAME}__menu-link-seperator`}>
+									<Separator />
+								</span>
+							) : null}
+						</>
 					))}
 			</nav>
 			<hr />
```
Comment (JackHowa) [inline blocks/links-bar-block/features/links-bar/default.jsx]: It sounds like this hr should be removed based on discussion. Then this will be handled with the layout

Comment (JackHowa) [reply blocks/links-bar-block/features/links-bar/default.jsx]: Actually, based on discussion, we want to leave it in. Then note the hr in the discrepancies doc

Comment (JackHowa) [reply blocks/links-bar-block/features/links-bar/default.jsx]: cc @RobinGiannattasio @matthewroach

Comment (RobinGiannattasio) [reply blocks/links-bar-block/features/links-bar/default.jsx]: Agreed! Leave it on and note it on the discrepancies doc.

**Review by ecarlisle — APPROVED** [review]

**Review by ecarlisle — CHANGES_REQUESTED** [review]
Sorry, meant to mark my existing change requests as resolved.

File: .storybook/themes/news.scss [hunk 26]
```diff
@@ -976,6 +985,15 @@
 						),
 					),
 				),
+				links-bar: (
+					display: flex,
```
Comment (matthewroach) [inline .storybook/themes/news.scss]: This is doing the same as the `<Stack>` component - can we use the `<Stack>` ?

Comment (matthewroach) [reply .storybook/themes/news.scss]: Also these styles are duplication of the styles set above - you only need to redefine the styles here **if** there are changes

File: blocks/links-bar-block/_index.scss (hunk 24 above)
Comment (JackHowa) [reply blocks/links-bar-block/_index.scss]: updated

**Comment by ecarlisle** [comment]
@malavikakoppula : 
It looks like you've covered all the issues I mentioned. I was however wondering if this block should be a `<Stack>` component instead of using `<nav>`. That may show more-consistent usage of our components across blocks and will use some existing CSS properties.

For example:
```
<Stack as="nav" justification="horizontal">
/* logic for `<Link>` and `<Separator>` */
</Stack>
```
(using the `as` property like that will set the top-level element of the `<Stack>` to `<nav>`.

cc: @JackHowa, @matthewroach for opinions on this. If we choose to go in that direction, I'm happy to help w/ any questions.

Thanks!
Eric

File: .storybook/themes/news.scss (hunk 26 above)
Comment (ecarlisle) [reply .storybook/themes/news.scss]: @matthewroach : Sorry, just noticed this comment. I have a duplicate suggestion below but put in some example code and reasoning. Thanks!

File: blocks/links-bar-block/features/links-bar/default.jsx (hunk 5 above)
Comment (JackHowa) [reply blocks/links-bar-block/features/links-bar/default.jsx]: this is fixed

File: .storybook/mock-content/linksBar.js (hunk 11 above)
Comment (JackHowa) [reply .storybook/mock-content/linksBar.js]: This is fixed

File: blocks/links-bar-block/features/links-bar/default.jsx (hunk 21 above)
Comment (JackHowa) [reply blocks/links-bar-block/features/links-bar/default.jsx]: this is fixed

File: blocks/links-bar-block/features/links-bar/default.jsx (hunk 25 above)
Comment (JackHowa) [reply blocks/links-bar-block/features/links-bar/default.jsx]: this is staying -- so resolved

File: .storybook/themes/news.scss (hunk 26 above)
Comment (JackHowa) [reply .storybook/themes/news.scss]: this looks like it's been resolved

File: blocks/links-bar-block/package.json [hunk 27]
```diff
@@ -29,8 +27,7 @@
   "peerDependencies": {
     "@arc-fusion/prop-types": "^0.1.5",
     "@wpmedia/engine-theme-sdk": "*",
```
Comment (ecarlisle) [inline blocks/links-bar-block/package.json]: Do we need this peer dependency?
cc: @JackHowa

File: .storybook/themes/news.scss [hunk 28]
```diff
@@ -976,6 +982,12 @@
 						),
 					),
 				),
+				links-bar: (
+					gap: var(--global-spacing-2),
+				),
+				links-bar-menu-link: (
+					text-decoration: none,
+				),
```
Comment (ecarlisle) [inline .storybook/themes/news.scss]: @malavikakoppula : Are these properties at `links-bar` and `links-bar-menu-link` needed at `desktop()` since they are set at `default()` and do not change?

Comment (malavikakoppula) [reply .storybook/themes/news.scss]: Styles are same for both `default() `as well as `desktop()`

image

File: blocks/links-bar-block/features/links-bar/default.jsx [hunk 29]
```diff
@@ -33,23 +32,36 @@ const LinksBar = ({ customFields: { navigationConfig = {}, ariaLabel } }) => {
...
-						<PrimaryFont as="span" className="links-menu" key={item._id}>
-							{item.node_type === "link" ? (
-								<Link href={item.url} name={item.display_name} />
-							) : (
-								<Link href={item._id} name={item.name} />
-							)}
-							{content.children.length !== index + 1 && showSeparator ? "\u00a0 • \u00a0" : ""}
-						</PrimaryFont>
+						<React.Fragment key={item._id}>
+							<span className={`${BLOCK_CLASS_NAME}__menu`} key={item._id}>
```
Comment (ecarlisle) [inline blocks/links-bar-block/features/links-bar/default.jsx]: @malavikakoppula : Was wondering if we need:
1. Both the keyed `React.Fragment` and the keyed `<span>`. They both seem to be serving the same purpose.
2. `${BLOCK_CLASS_NAME}__menu`, as we're not applying any styles to it.

**Comment by ecarlisle** [comment]
@malavikakoppula : The only other link I can't figure out is why I'm seeing menu items in a Serif font. I would expect this to inherit the Sans-Serif font from `<body>`. 🤔 
image

File: .storybook/themes/news.scss (hunk 28 above)
Comment (ecarlisle) [reply .storybook/themes/news.scss]: Yes. The CSS is mobile-first, so `desktop()` will use the values in `default()` (default being "mobile").

**Comment by malavikakoppula** [comment]
> @malavikakoppula : The only other link I can't figure out is why I'm seeing menu items in a Serif font. I would expect this to inherit the Sans-Serif font from `<body>`. 🤔 image

Did not added any specific styles for this. In my local it Looks like this

Screen Shot 2022-06-09 at 1 54 39 PM

Am I missing somethig?

File: blocks/links-bar-block/package.json (hunk 27 above)
Comment (nschubach) [reply blocks/links-bar-block/package.json]: This should be removed.

**Comment by nschubach** [comment]
Remove the engine-theme-sdk from the tests please.  (Can't comment on the exact lines because github):
```
jest.mock("@wpmedia/engine-theme-sdk", () => ({
	formatURL: jest.fn((input) => input.toString()),
}));
```

**Comment by malavikakoppula** [comment]
> Remove the engine-theme-sdk from the tests please. (Can't comment on the exact lines because github):
> 
> ```
> jest.mock("@wpmedia/engine-theme-sdk", () => ({
> 	formatURL: jest.fn((input) => input.toString()),
> }));
> ```
removed it

File: blocks/footer-block/features/footer/default.jsx [hunk 30]
```diff
@@ -6,7 +6,7 @@ import { useFusionContext } from "fusion:context";
 import getProperties from "fusion:properties";
 import getTranslatedPhrases from "fusion:intl";
 import getThemeStyle from "fusion:themes";
-import Link from "@wpmedia/links-bar-block";
+// import Link from "@wpmedia/links-bar-block";
```
Comment (nschubach) [inline blocks/footer-block/features/footer/default.jsx:9]:
```suggestion
```

File: blocks/links-bar-block/features/links-bar/default.jsx [hunk 31]
```diff
@@ -33,23 +32,34 @@ const LinksBar = ({ customFields: { navigationConfig = {}, ariaLabel } }) => {
...
-				className="links-bar"
 				aria-label={ariaLabel || phrases.t("links-bar-block.element-aria-label")}
 			>
 				{menuItems &&
 					menuItems.map((item, index) => (
-						<PrimaryFont as="span" className="links-menu" key={item._id}>
+						<React.Fragment key={item._id}>
 							{item.node_type === "link" ? (
-								<Link href={item.url} name={item.display_name} />
+								<Link href={item.url} className={`${BLOCK_CLASS_NAME}__menu-link`}>
```
Comment (nschubach) [inline blocks/links-bar-block/features/links-bar/default.jsx]: If there is only ever one link style in this block, you can style the link with the component child selector in the tokens:
```
    blocks: (
        links-bar: (
            components: (
                link: (
                    color: red,
                )
            )
        ),
    ),
```
Adding a class `__menu-link` is redundant.

Comment (nschubach) [reply blocks/links-bar-block/features/links-bar/default.jsx]: So you should not need to add this class.

File: blocks/links-bar-block/features/links-bar/default.jsx [hunk 32]
```diff
@@ -33,23 +32,34 @@ const LinksBar = ({ customFields: { navigationConfig = {}, ariaLabel } }) => {
...
 							) : (
-								<Link href={item._id} name={item.name} />
+								<Link href={item._id} className={`${BLOCK_CLASS_NAME}__menu-link`}>
+									{item.name}
+								</Link>
 							)}
-							{content.children.length !== index + 1 && showSeparator ? "\u00a0 • \u00a0" : ""}
-						</PrimaryFont>
+							{content.children.length !== index + 1 && showSeparator ? (
+								<span className={`${BLOCK_CLASS_NAME}__menu-link-seperator`}>
```
Comment (nschubach) [inline blocks/links-bar-block/features/links-bar/default.jsx]: Span is not needed, and Separator styling can be defined like the Link above.

Comment (nschubach) [reply blocks/links-bar-block/features/links-bar/default.jsx]: \<Separator \/\> returns a span already.

File: blocks/links-bar-block/features/links-bar/default.jsx [hunk 33]
```diff
@@ -33,23 +32,34 @@ const LinksBar = ({ customFields: { navigationConfig = {}, ariaLabel } }) => {
 
 	return (
```
Comment (nschubach) [inline blocks/links-bar-block/features/links-bar/default.jsx:33]: Most likely need a check here to render this only if `menuItems?.length > 0`?

File: blocks/links-bar-block/features/links-bar/default.jsx [hunk 34]
```diff
@@ -33,23 +32,34 @@ const LinksBar = ({ customFields: { navigationConfig = {}, ariaLabel } }) => {
...
 							)}
-							{content.children.length !== index + 1 && showSeparator ? "\u00a0 • \u00a0" : ""}
-						</PrimaryFont>
+							{content.children.length !== index + 1 && showSeparator ? (
+								<span className={`${BLOCK_CLASS_NAME}__menu-link-seperator`}>
+									<Separator />
+								</span>
+							) : null}
+						</React.Fragment>
 					))}
```
Comment (nschubach) [inline blocks/links-bar-block/features/links-bar/default.jsx]: With the above noted changes, this can be simplified to:
```
				{menuItems.map((item, index) => (
					<React.Fragment key={item._id}>
						{index > 0 && showSeparator ? <Separator /> : null}
						{item.node_type === "link" ? (
							<Link href={item.url}>{item.display_name}</Link>
						) : (
							<Link href={item._id}>{item.name}</Link>
						)}
					</React.Fragment>
				))}
```

File: blocks/links-bar-block/features/links-bar/default.jsx (hunk 31 above)
Comment (nschubach) [reply blocks/links-bar-block/features/links-bar/default.jsx]: Seems like this is not using the Components Link... I commented on that above.  If we can use that, it makes styling this much easier.

**Review by nschubach — CHANGES_REQUESTED** [review]
Just a couple concerns.
* redundant links class (should just use theme nesting for component styling)
* redundant span around separator (can be styled via the component nesting like the links)
* length check for not rendering an empty stack

File: blocks/footer-block/features/footer/_children/link.jsx [hunk 35]
```diff
@@ -0,0 +1,52 @@
...
+		<a href={fixTrailingSlash(href)}>{name}</a>
+	);
+};
+
+Link.propTypes = {
+	href: PropTypes.string,
+	name: PropTypes.string,
+};
+
+export default Link;
```
[the remark this answers is not in the document]
Comment (nschubach) [reply blocks/footer-block/features/footer/_children/link.jsx:52]: ignore this

File: blocks/links-bar-block/features/links-bar/default.jsx [hunk 36]
```diff
@@ -33,23 +32,26 @@ const LinksBar = ({ customFields: { navigationConfig = {}, ariaLabel } }) => {
...
+				className={BLOCK_CLASS_NAME}
+				justification="center"
+				direction="row"
+				as="nav"
 				key={id}
-				className="links-bar"
 				aria-label={ariaLabel || phrases.t("links-bar-block.element-aria-label")}
 			>
-				{menuItems &&
+				{menuItems?.length > 0 &&
```
Comment (nschubach) [inline blocks/links-bar-block/features/links-bar/default.jsx]: This should be moved up to encompass the whole return.  There's no need to render the stack or `<hr />` if there are no items, right?

**Review by nschubach — CHANGES_REQUESTED** [review]
Small change to restrict rendering if there are no items.

File: blocks/links-bar-block/features/links-bar/default.jsx (hunk 33 above)
Comment (JackHowa) [reply blocks/links-bar-block/features/links-bar/default.jsx:33]: @malavikakoppula does this make sense? we can also pair on this

File: blocks/links-bar-block/features/links-bar/default.jsx (hunk 36 above)
Comment (JackHowa) [reply blocks/links-bar-block/features/links-bar/default.jsx]: looks like this needs to move up @malavikakoppula based on the comment. wanted to make sure this made sense https://github.com/WPMedia/arc-themes-blocks/pull/1357/files#r893897519

Comment (malavikakoppula) [reply blocks/links-bar-block/features/links-bar/default.jsx]: updated empty check to hr tag while no data

File: blocks/links-bar-block/features/links-bar/default.jsx (hunk 33 above)
Comment (malavikakoppula) [reply blocks/links-bar-block/features/links-bar/default.jsx:33]: Yes Jack, Updated it

**Review by JackHowa — APPROVED** [review]
Looks good based on Nick's change!

**Review by nschubach — APPROVED** [review]
👍

**Comment by JackHowa** [comment]
Hmm baseline changes due to the font change?

**Comment by JackHowa** [comment]
Looks like the Abel change to the preview head is causing chromatic to list multiple regressions. Can you undo the change to the preview head @malavikakoppula

**Comment by nschubach** [comment]
I thought the font change was intentional?

**Comment by malavikakoppula** [comment]
> I thought the font change was intentional?

Yes, it was intentional that primary family is assigned as "Abel" , not mentioned in preview-head

**Comment by JackHowa** [comment]
It's not in the base branch, right?  https://github.com/WPMedia/arc-themes-blocks/blob/arc-themes-release-version-2.0.1/.storybook/preview-head.html

**Comment by JackHowa** [comment]
Oh ok

**Review by ecarlisle — APPROVED** [review]
